programs/mango-v4/src/instructions/account_buyback_fees_with_mngo.rs

```diff
@@ -34,21 +34,36 @@ pub fn account_buyback_fees_with_mngo(
     let mut mngo_bank = ctx.accounts.mngo_bank.load_mut()?;
     let mut fees_bank = ctx.accounts.fees_bank.load_mut()?;
 
+    let clock = Clock::get()?;
+    let now_ts = clock.unix_timestamp.try_into().unwrap();
+    let slot = clock.slot;
+
+    let mngo_oracle_price = mngo_bank.oracle_price(
+        &AccountInfoRef::borrow(&ctx.accounts.mngo_oracle.as_ref())?,
+        Some(slot),
+    )?;
+    let mngo_asset_price = mngo_oracle_price.min(mngo_bank.stable_price());
+
+    let fees_oracle_price = fees_bank.oracle_price(
+        &AccountInfoRef::borrow(&ctx.accounts.fees_oracle.as_ref())?,
+        Some(slot),
+    )?;
+    let fees_liab_price = fees_oracle_price.max(fees_bank.stable_price());
+
     let bonus_factor = I80F48::from_num(group.buyback_fees_mngo_bonus_factor);
 
-    let now_ts = Clock::get()?.unix_timestamp.try_into().unwrap();
     account
         .fixed
         .expire_buyback_fees(now_ts, group.buyback_fees_expiry_interval);
 
     // quick return if nothing to buyback
-    let mut max_buyback = {
+    let mut max_buyback_usd_native = {
```

Comment:
I thought we agreed that if no suffix then its native

Reply:
right, I just did it because the other fields had a _native postfix in that function. Maybe I'll get rid of all of them.